test/test_testing.py

```diff
@@ -1147,5 +1147,69 @@ def test_mismatching_device_no_check(self, devices):
 instantiate_device_type_tests(TestAssertsMultiDevice, globals(), only_for="cuda")
 
 
+class TestAssertsSparse(TestCase):
+    def test_matching(self):
+        indices = (
+            (0, 1),
+            (1, 0),
+        )
+        values = (1, 2)
+        actual = torch.sparse_coo_tensor(indices, values, (2, 2))
+        expected = actual.clone()
+
+        for fn in assert_fns_with_inputs(actual, expected):
+            fn()
+
+    def test_matching_uncoalesced(self):
+        indices = (
+            (0, 1, 0,),
+            (1, 0, 1,),
+        )
+        values = (1, 1, 1)
+        actual = torch.sparse_coo_tensor(indices, values, (2, 2))
+        expected = actual.clone().coalesce()
+
+        for fn in assert_fns_with_inputs(actual, expected):
+            fn()
+
+    def test_mismatching_indices_msg(self):
```

Comment:
Requiring indices match seems like a neat way to sidestep problems like: "what if one sparse tensor specifies a zero explicitly but the other sparse tensor doesn't?"

If sparse tensors have zeros for unspecified values then those tensors are mathematically equivalent, but it seems like this is treating the indices tensor as metadata (which we require to match).